docs/rules/amd-function-arity.md

```diff
@@ -0,0 +1,242 @@
+# Ensure AMD callbacks have correct number of parameters (amd-function-arity)
+
+
+## Rule Details
+
+This rule aims to ensure that AMD callbacks have the correct number of parameters in order to avoid confusion and maintainability issues.
+
+### Options
+
+This rule takes one option, an object with the key `allowExtraDependencies` and a Boolean value (default is `false`).
+
+```json
+    "requirejs/enforce-define": [2, { "allowExtraDependencies": true }]
```

Comment:
Also I'm not 100% sold on the naming of this option. It's more about allowing require of dependencies for side effects, no?